### PR TITLE
Added authorization by role for controllers actions

### PR DESCRIPTION
**Pull request by Igor-Polatajko**

### CODE REVIEW

File: src/main/java/com/smartcity/controller/TaskController.java [hunk 1]
```diff
@@ -45,6 +48,7 @@ public TaskDto updateTask(
         return taskService.update(taskDto);
     }
 
+    @PreAuthorize("hasAnyRole('RESPONSIBLE_PERSON', 'ROLE_SUPERVISOR')")
```
Comment (iholovatyi) [inline src/main/java/com/smartcity/controller/TaskController.java]: Do we want to allow Responsible person to delete tasks?

File: src/main/java/com/smartcity/controller/TransactionController.java [hunk 2]
```diff
@@ -23,12 +24,14 @@ public TransactionController(TransactionService transService) {
         this.transService = transService;
     }
 
+    @PreAuthorize("hasAnyRole('ROLE_SUPERVISOR', 'ROLE_ADMIN')")
```
Comment (iholovatyi) [inline src/main/java/com/smartcity/controller/TransactionController.java]: Why we don't want to show transactions to a responsible person?

File: src/main/java/com/smartcity/controller/TransactionController.java [hunk 3]
```diff
@@ -39,18 +42,21 @@ public TransactionDto updateTransaction(
...
     @ResponseStatus(HttpStatus.OK)
     @DeleteMapping("/{id}")
     public boolean deleteTransaction(@PathVariable("id") Long id) {
         return transService.delete(id);
     }
 
+    @PreAuthorize("hasAnyRole('ROLE_SUPERVISOR', 'ROLE_ADMIN')")
```
Comment (iholovatyi) [inline src/main/java/com/smartcity/controller/TransactionController.java]: Why we don't want to show transactions to a responsible person?

File: src/main/java/com/smartcity/controller/UserController.java [hunk 4]
```diff
@@ -26,12 +23,14 @@ public UserController(UserService userService) {
         this.userService = userService;
     }
 
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
```
Comment (iholovatyi) [inline src/main/java/com/smartcity/controller/UserController.java]: I don't think we need to restrict viewing users for specific role. Just remove this line

File: src/main/java/com/smartcity/controller/UserController.java [hunk 5]
```diff
@@ -26,12 +23,14 @@ public UserController(UserService userService) {
...
     @ResponseStatus(HttpStatus.OK)
     @GetMapping(value = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
     public UserDto getById(@PathVariable("id") Long id) {
         return userService.get(id);
     }
 
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
```
Comment (iholovatyi) [inline src/main/java/com/smartcity/controller/UserController.java]: Remove this as well

File: src/main/java/com/smartcity/controller/UserController.java [hunk 6]
```diff
@@ -45,6 +44,7 @@ public UserDto updateUser(@Validated(ExistingRecord.class) @PathVariable("id") L
         return userService.update(userDto);
     }
 
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
```
Comment (iholovatyi) [inline src/main/java/com/smartcity/controller/UserController.java]: Remove role "USER_ROLE", only admin should be able to delete users

File: src/main/java/com/smartcity/controller/OrganizationController.java [hunk 7]
```diff
@@ -9,13 +9,15 @@
...
 import org.springframework.http.MediaType;
+import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.validation.annotation.Validated;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
 
 @RestController
 @RequestMapping("/organizations")
+@PreAuthorize("hasRole('ROLE_ADMIN')")
```
Comment (iholovatyi) [inline src/main/java/com/smartcity/controller/OrganizationController.java]: I think this is redundant as you annotated all the methods with some roles no need to annotate controller in that case

File: src/main/java/com/smartcity/controller/TaskController.java [hunk 8]
```diff
@@ -23,6 +24,7 @@ public TaskController(TaskService taskService) {
         this.taskService = taskService;
     }
 
+    @PreAuthorize("hasAnyRole('ROLE_SUPERVISOR')")
```
Comment (iholovatyi) [inline src/main/java/com/smartcity/controller/TaskController.java]: No point to use hasAnyRole for one role. Use
`@PreAuthorize("hasRole('ROLE_SUPERVISOR')")`

File: src/main/java/com/smartcity/controller/OrganizationController.java (hunk 7 above)
Comment (Igor-Polatajko) [reply src/main/java/com/smartcity/controller/OrganizationController.java]: Sorry, I missed it. I had to delete it.

**Pull request merged**